examples/tutorials/analysis-3d/event_sampling_nrg_depend_models.py

```diff
@@ -0,0 +1,384 @@
+# time. Here we adopt an `oversample_energy_factor` of 10:
+#
+
+sampler = MapDatasetEventSampler(random_state=0, oversample_energy_factor=10)
+events = sampler.run(dataset, observation)
+
+######################################################################
+# Let's inspect the simulated events in the source region:
+#
+
+src_position = SkyCoord(100.0, 30.0, frame="icrs", unit="deg")
+
+on_region_radius = Angle("0.15 deg")
+on_region = CircleSkyRegion(center=src_position, radius=on_region_radius)
+
+src_events = events.select_region(on_region)
+
+src_events.peek()
+
+
+######################################################################
+# Fit the simulated data
+# ~~~~~~~~~~~~~~~~~~~~~~
+#
+# We fit the events in one time bin, adopting a model given by a
+# powerlaw spectral model, with no temporal information and point-like
+# morphology. We fix the source coordinates during the fit for simplicity:
+#
+
+spec = PowerLawSpectralModel(
+    index=2, amplitude=5e-11 * u.cm**-2 * u.s**-1 * u.TeV**-1, reference="1 TeV"
+)
+
+spatial_model = PointSpatialModel(lon_0="100 deg", lat_0="30 deg", frame="icrs")
+
+model = SkyModel(
+    spatial_model=spatial_model,
+    spectral_model=spec,
+    name="fit_source",
+)
+
+model.spatial_model.parameters[0].frozen = True
+model.spatial_model.parameters[1].frozen = True
+
+bkg_model = FoVBackgroundModel(dataset_name="my-dataset")
+
+model_fit = [model, bkg_model]
+
+
+######################################################################
+# Let's create a new observation, with the time bin starting
+# at 360s and with a duration of 100s, and a new dataset for
+# the fit:
+#
+
+livetime = 100 * u.s
+
+i = 0
+tstart = t_ref + (time_min[i] + 360 * u.s).to("d")
+observation = Observation.create(
+    obs_id=1001,
+    pointing=pointing,
+    livetime=livetime,
+    irfs=irfs,
+    location=location,
+    tstart=tstart,
+)
+
+dataset_fit = maker.run(empty, observation)
+
+######################################################################
+# We select only the events in the time bin of reference, and we
+# fill the dataset with them.
+#
+
+filt_evt = events.select_time([observation.gti.time_start, observation.gti.time_stop])
+counts = Map.from_geom(geom)
+counts.fill_events(filt_evt)
+dataset_fit.counts = counts
+
+dataset_fit.models = model_fit
+print(dataset_fit.models)
+
+
+######################################################################
+# Let's fit the dataset
+#
+
+fit = Fit()
+result = fit.run(dataset_fit)
+print(result)
+
+result.parameters.to_table()
+
+######################################################################
+# The best-fit spectral parameters are similar but not exactly equal
+# to the simulated ones. This is due to interpolation method applied
+# to the temporal model!
+#
+
+######################################################################
+# Exercises
+# ---------
+#
+# -  Try to create a temporal model with a more complex energy-dependent
+#    evolution;
```

Comment:
Maybe suggest to read the template model in gammapy data and perform a simulation?

Reply:
Ok.